src/Widgets/AgendaEventRow.vala

```diff
@@ -35,12 +35,61 @@ public class Maya.View.AgendaEventRow : Gtk.ListBoxRow {
     public bool is_multiday { public get; private set; default = false; }
     public Gtk.Revealer revealer { public get; private set; }
 
+    private Gtk.Image event_image;
     private Gtk.Label name_label;
     private Gtk.Label datatime_label;
     private Gtk.Label location_label;
     private Gtk.StyleContext event_image_context;
     private Gtk.StyleContext main_grid_context;
 
+    private enum Category {
+        NONE,
+        APPOINTMENT,
+        CELEBRATION,
+        CALL,
+        DRINKS,
+        DRIVING,
+        FLIGHT,
+        FOOD,
+        LEGAL,
+        MOVIE,
+        WEDDING,
+        N_CATEGORIES
```

Comment:
```suggestion
        N_CATEGORIES;
        public unowned string get_builtin_keywords () {
            switch (this) {
                …
            }
        }
```
Doing this would probably simplify the code afterwards

Reply:
Thanks, that is much better.  I have also done the same for the icon name since there is only one icon per category,